### DISCUSSION TITLE
Changing Directory where photos are saved

### DISCUSSION THREAD

**Discussion by hawkinle** [geezacoleman/OpenWeedLocator#119] [Q&A] [answered]
Long time listener first time poster :D 

I was just wondering is there anyway for me to be able to change the default image save location to either USB or NVME drive on the PI ? So that we can save storage on the PI and also save processing power .  thanks

**Comment by geezacoleman** [comment, marked as answer]
Thanks for posting! 

For the USB location, you'll just need to change[ L573 in `owl.py`](https://github.com/geezacoleman/OpenWeedLocator/blob/962e3fbf6a9e7a1be822a516436cb5b7f8a3e2e0/owl.py#L573C14-L573C37) to whatever directory/USB drive you're saving to. As default it currents just saves to this directory `saveDir='images/bbox2',` but if you add a USB it should appear in the home/pi/media/YOUR_USB_DRIVE directory. Double check this is working first with a screen - we've had issues in the past saving to a USB drive.

NVME is a great idea! I don't have any experience with that though, but it seems there are a few different tutorials out there on how to boot from one, so presumably how to save to one wouldn't be too tricky.

Hopefully this helps - let me know if you run into any issues.

**Reply by hawkinle** [reply]
Muy Excellente ,  thats most helpful.  I appreciate  it. 



I think with the NVME ill just install it on as a shared drive and then it can be a storage hub.

**Reply by geezacoleman** [reply]
No worries! I just realised I typed `home/pi/media` which is the incorrect path to the default USB mount location. Should be `media/pi/USB_NAME`. I did some more reading, and [this forum ](https://forums.raspberrypi.com/viewtopic.php?t=48958)is quite helpful - I think some of the issues we've had were about user group permissions and accessing the drive.

Would be good to hear how you go with an NVME!
